LibOS/shim/include/shim_internal.h

```diff
@@ -446,6 +448,8 @@ static inline void disable_preempt (shim_tcb_t * tcb)
 static inline void __enable_preempt (shim_tcb_t * tcb)
 {
     //tcb->context.syscall_nr -= SYSCALL_NR_PREEMPT_INC;
+    /* Assert if this counter underflows */
+    assert(tcb->context.preempt > 0);
```

Comment:
Isn't preempt is an unsigned type?  If you only care about 0 this is fine, but if you want to check for a negative value, this needs a cast.